### DISCUSSION TITLE
How to make property take precedence over indexer.

### DISCUSSION THREAD

**Discussion by david-w-lee** [sebastienros/jint#1462] [Q&A]
Let's say I have a class which contains indexer and property at the same time like below.
I want to get value from MyProp when I call myobj.MyProp but I want to get other values from indexer.
But when I run the following code, I get value from indexer instead of myobj.MyProp.
How can I fix this?

```c#
    public class MyClass
    {
        public string this[string propname]
        {
            get { return "from indexer";  }
        }

        public string MyProp
        {
            get { return "from property"; }
        }
    }

    public class Program
    {
        public static void Main()
        {
            var engine = new Engine();
            engine.SetValue("myobj", new MyClass());
            engine.SetValue("debug", new Action<string>(Console.WriteLine));
            var res = engine.Evaluate("debug(myobj.MyProp)");
            Console.ReadKey();
        }
    }
```

**Comment by lahma** [comment]
The behavior comes from here:

https://github.com/sebastienros/jint/blob/4e1893d4696bde1c36a08deea85a1546bc316aaf/Jint/Runtime/Interop/Reflection/ReflectionAccessor.cs#L43-L58

So for some reason the indexer is preferred first. If you can create a test case that shows fixed priority usage, create required tweak to this code logic and other interop tests still pass, a pull request would be appreciated.

**Comment by lahma** [comment]
I re-checked this and this is just too ambiguous to handle. Your indexer should return null when "MyProp" is requested, otherwise indexer wins as Jint doesn't know which one you would like to use anyway.
